llarp/util/compare_ptr.hpp

```diff
@@ -16,4 +17,16 @@ namespace llarp
       return Compare()(left, right);
     }
   };
+
+  /// type for comparing weak_ptr
```

Comment:
```suggestion
  /// type for comparing weak_ptr by value
```
(to disambiguate from std::owner_less, which gives comparison by owned pointer).